### ISSUE TITLE
ngrx store can't save the $event.file

### ISSUE THREAD

**Issue by Danieliverant** [aberezkin/ng2-image-upload#181]
I'm using NGRX store 4 with my Angular 5 project,
and when I'm sending the` $event.file` to the reducer it appears empty.
(i'm using the onUploadFinished method)
also, if I send the whole `$event` I will see in the store the` .src` but the `.file` will be an empty object.

any idea what could be the problem?

**Comment by reenezden** [comment]
Yes, I'm also experiencing this issue. I can only get a base64 encoded string in src and file{} is empty

**Comment by Danieliverant** [comment]
I figured out that if you use "event" instead of "$event" in your js/ts code that works.
in the HMTL use onUploadFinished($event) and in your JS use onUploadFinished(event) 
and then to get the file use event.file.

**Comment by reenezden** [comment]
Thanks for the reply, I'm definetely checkin it out tonight

**Comment by reenezden** [comment]
@Danieliverant I tried your solution but I'm still getting an empty object, I can get the image name though when I write: event.file.name but event.file shows {} an empty object. Weird

**Comment by Danieliverant** [comment]
@reenezden  try to create new object before you save to the store.

`  onUploadFinished(event: FileHolder) {
    const file = {
      uniqID: this.uniqID,
      name: event.file.name,
      size: event.file.size,
      type: event.file.type,
      src: event.src
    };
    this.store.dispatch(new fromStore.SaveImage(file));`

**Comment by reenezden** [comment]
Thanks alot @Danieliverant. I'm back from my trip. Going to try this out

**Comment by Danieliverant** [comment]
@reenezden  
better solution:

add [url]="" in your HTML <image-upload> and than you can use event.file

**Comment by reenezden** [comment]
Thanks alot @Danieliverant. The worked well 100%
